core/includes/theme.inc

```diff
@@ -1730,24 +1730,25 @@ function theme_status_messages($variables) {
     'status' => t('Status message'),
     'error' => t('Error message'),
     'warning' => t('Warning message'),
+    'info' => t('Info message'),
   );
 
   foreach ($message_types as $type => $messages) {
     $output .= "<div class=\"messages $type\">\n";
     if (!empty($status_heading[$type])) {
-      $output .= '<h2 class="element-invisible">' . $status_heading[$type] . "</h2>\n";
+      $output .= '  <h2 class="element-invisible">' . $status_heading[$type] . "</h2>\n";
     }
     if (count($messages) > 1) {
-      $output .= " <ul>\n";
+      $output .= "  <ul>\n";
       foreach ($messages as $message) {
-        $output .= '  <li>' . $message . "</li>\n";
+        $output .= "    <li>" . $message . "</li>\n";
       }
-      $output .= " </ul>\n";
+      $output .= "  </ul>";
     }
     else {
       $output .= reset($messages);
     }
-    $output .= "</div>\n";
+    $output .= "\n</div>\n";
```

Comment:
Nitpicking : `\n` are always printed out at the end of every line, so maybe restore this `\n` after the `</ul>` closing tag above.

Reply:
This was intentional @opi ...you see, all these `\n`s and additional spaces that's going on here is in order to print the entire message `<div>` in a pretty format when one inspects the rendered code. With multiple messages of the same type, having the leading `\n` either way, renders things like so:

```
<div class="messages info">
  <h2 class="element-invisible">Info message</h2>
  <ul>
    <li>info1</li>
    <li>info2</li>
  </ul>
</div>
```

...so no problem either way. But when you have single messages, it prints them like so:

- with the `\n` after the `</ul>`:

```
<div class="messages info">
  <h2 class="element-invisible">Info message</h2>
single info message</div>
```

- with the `\n` before the `</div>`:

```
<div class="messages info">
  <h2 class="element-invisible">Info message</h2>
single info message
</div>
```

This is because moving the `\n` after the `</ul>` puts it inside the `if` statement, which is not true for single messages of the same type. This effectively makes it so that the `\n` is not printed in these cases, which makes the closing `</div>` tag print in the same line as the actual message text, which defeats the purpose of using all these `\n`s (which was to make the rendered code look pretty).

I hope that this makes sense @opi 

PS: In general, my preference would be to remove all this "witchcraft" from specific places like this, and have the entire code that is generated by Drupal/Backdrop be prettified on the fly, at a lower-level. But I guess that if possible, someone would have done it by now 😄

Reply:
wow, such an explanation ! And +1 for removing this tabs and line from the code. I never read raw sources anymore, browsers inspector always prettify the html now.